nipst/nipst.go

```diff
@@ -244,82 +209,36 @@ func (nb *NIPSTBuilder) BuildNIPST(challenge *common.Hash) (*NIPST, error) {
 			"(service id: %v, round id: %v)",
 			nb.poetProver.id(), round.Id)
 
-		nb.nipst.NipstChallenge = poetChallenge
-		nb.nipst.PoetRound = round
-		nb.nipst.persist()
+		nipst.NipstChallenge = poetChallenge
+		nb.state.PoetRound = round
+		nb.state.persist()
 	}
 
-	// Phase 1: Wait for PoET service round membership proof.
-	if nb.nipst.PoetMembershipProof == nil {
-		nb.log.Info("querying round membership proof from PoET proving service "+
-			"(service id: %v, round id: %v, challenge: %x)",
-			nb.poetProver.id(), nb.nipst.PoetRound.Id, nb.nipst.NipstChallenge)
-
-		mproof, err := nb.poetProver.subscribeMembershipProof(nb.nipst.PoetRound, *nb.nipst.NipstChallenge, defTimeout)
+	// Phase 1: receive proofs from PoET service
+	if nb.state.PoetProofRoot == nil {
+		poetProofRoot, err := nb.poetDb.GetPoetProofRoot(nb.state.PoetId, nb.state.PoetRound)
 		if err != nil {
-			return nil, fmt.Errorf("failed to receive PoET round membership proof: %v", err)
+			// TODO(noamnelke): handle timeout
```

Comment:
Any reason why not do that as part of this PR?

Reply:
This is not as simple as it may seem. I'm not talking about a timeout in seconds, but rather estimating how long the poet is supposed to run, adding some padding to account for possible propagation and other delays and if that time has passed assume the poet is dead, blacklist it, scrap the current ATX and use a different poet (a bunch of mechanisms we don't have yet).

Reply:
ok, we'll tackle this as part of mining recovery